read_excel tests sporadically fail in CI because some file handles remain open

### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux
- **Modin version** (`modin.__version__`): master (e.g. 3e62906)
- **Python version**: any
- **Code we can use to reproduce**: watch for `read_io` tests

<!--
You can obtain the Modin version with

python -c "import modin; print(modin.__version__)"
-->

### Describe the problem
<!-- Describe the problem clearly here. -->
Something seems to be reading statistics from time to time, as tests that fail are always having files from `/proc/<pid>/<something>` left open.

### Source code / logs
<!-- Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem. -->
<details><summary>Example: https://github.com/modin-project/modin/runs/1572578393#step:22:64</summary>

```
_____________________ TestExcel.test_read_excel_index_col ______________________

a = (<modin.pandas.test.test_io.TestExcel object at 0x7f0df11ea9d0>,), kw = {}
fstart = [], leaks = [('/proc/10921/stat', 60)], item = ('/proc/10921/stat', 60)

    @functools.wraps(func)
    def check(*a, **kw):
        fstart = _get_open_files()
        try:
            return func(*a, **kw)
        finally:
            leaks = []
            for item in _get_open_files():
                try:
                    fstart.remove(item)
                except ValueError:
                    leaks.append(item)
            assert (
                not leaks
>           ), f"Unexpected open handles left for: {', '.join(item[0] for item in leaks)}"
E           AssertionError: Unexpected open handles left for: /proc/10921/stat
```

</details>

More info: the check was introduced in https://github.com/modin-project/modin/pull/2514

--- a/modin/pandas/test/utils.py
+++ b/modin/pandas/test/utils.py
@@ -1047,7 +1047,10 @@ def check_file_leaks(func):
                 try:
                     fstart.remove(item)
                 except ValueError:
-                    leaks.append(item)
+                    # ignore files in /proc/, as they have nothing to do with
+                    # modin reading any data (and this is what we care about)
+                    if not item[0].startswith("/proc/"):
+                        leaks.append(item)
             assert (
                 not leaks
             ), f"Unexpected open handles left for: {', '.join(item[0] for item in leaks)}"

